code/modules/antagonists/heretic/status_effects/void_chill.dm

```diff
@@ -0,0 +1,113 @@
+/*!
+ * Contains the "Void Chill" status effect. Harmful debuff which freezes and slows down non-heretics
+ * Cannot affect silicons (How are you gonna freeze a robot?)
+ */
+/datum/status_effect/void_chill
+	id = "void_chill"
+	duration = 30 SECONDS
+	alert_type = /atom/movable/screen/alert/status_effect/void_chill
+	status_type = STATUS_EFFECT_REFRESH //Custom code
+	on_remove_on_mob_delete = TRUE
+	remove_on_fullheal = TRUE
+	///Current amount of stacks we have
+	var/stacks
+	///Maximum of stacks that we could possibly get
+	var/stack_limit = 5
+	///icon for the overlay
+	var/image/stacks_overlay
+
+/datum/status_effect/void_chill/on_creation(mob/living/new_owner, new_stacks, ...)
+	. = ..()
+	RegisterSignal(owner, COMSIG_ATOM_UPDATE_OVERLAYS, PROC_REF(update_stacks_overlay))
+	set_stacks(new_stacks)
+	owner.add_atom_colour(COLOR_BLUE_LIGHT, TEMPORARY_COLOUR_PRIORITY)
+	owner.update_icon(UPDATE_OVERLAYS)
+
+/datum/status_effect/void_chill/on_apply()
+	if(issilicon(owner))
+		return FALSE
+	return TRUE
+
+/datum/status_effect/void_chill/on_remove()
+	owner.update_icon(UPDATE_OVERLAYS)
+	owner.remove_atom_colour(TEMPORARY_COLOUR_PRIORITY, COLOR_BLUE_LIGHT)
+	owner.remove_movespeed_modifier(/datum/movespeed_modifier/void_chill)
+	owner.remove_alt_appearance("heretic_status")
+	REMOVE_TRAIT(owner, TRAIT_HYPOTHERMIC, REF(src))
+	UnregisterSignal(owner, COMSIG_ATOM_UPDATE_OVERLAYS)
+
+/datum/status_effect/void_chill/tick(seconds_between_ticks)
+	owner.adjust_bodytemperature(-12 * stacks)
```

Comment:
Just in case someone changes the tick interval later carelessly
(By default it's 1 so it really does nothing rn)
```suggestion
	owner.adjust_bodytemperature(-12 * stacks * seconds_between_ticks)
```